### PR TITLE
Job sequencing with deadlines

### PR DESCRIPTION
**Pull request by lakshayroop5**
### Describe your change:
In this PR, I have written an optimized algorithm for the issue "Job Sequencing with Deadlines". This problem required a greedy approach so I have created a new python file "job_sequencing_with_deadlines" in the "greedy_methods" directory.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
Click `Details` next to the failing tests below to see:
```
greedy_methods/job_sequencing_with_deadline.py:10:1: E302 expected 2 blank lines, found 1
greedy_methods/job_sequencing_with_deadline.py:23:89: E501 line too long (90 > 88 characters)
greedy_methods/job_sequencing_with_deadline.py:25:89: E501 line too long (103 > 88 characters)
```

File: greedy_methods/job_sequencing_with_deadline.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+"""
+This is a pure Python implementation of the greedy-merge-sort algorithm
+
+<<<<<<< HEAD
```
Comment (cclauss) [inline greedy_methods/job_sequencing_with_deadline.py:4]: This is a Python syntax error.

**Review by lakshayroop5 — COMMENTED** [review]
corrected code

**Review by lakshayroop5 — COMMENTED** [review]
corrected spacing issue

**Review by lakshayroop5 — COMMENTED** [review]
corrected spacing issue

**Comment by lakshayroop5** [comment]
I hope the algorithm written in this PR solves the given problem optimally and accurately. Kindly review my PR as soon as possible and suggest any necessary changes if required.
Thank you

File: greedy_methods/job_sequencing_with_deadline.py [hunk 2]
```diff
@@ -20,9 +20,11 @@ def job_sequencing_with_deadlines(num_jobs: int, jobs: list) -> list:
         in a given time frame
 
     Examples:
-    >>> job_sequencing_with_deadlines(4, [(1, 4, 20), (2, 1, 10), (3, 1, 40), (4, 1, 30)])
+    >>> job_sequencing_with_deadlines(4, [(1, 4, 20), (2, 1, 10), \
+    (3, 1, 40), (4, 1, 30)])
```
Comment (cclauss) [inline greedy_methods/job_sequencing_with_deadline.py:24]:
```suggestion
    >>> job_sequencing_with_deadlines(4, [(1, 4, 20), (2, 1, 10), 
    ... (3, 1, 40), (4, 1, 30)])
```

**Review by lakshayroop5 — COMMENTED** [review]
updated reviews

**Review by cclauss — APPROVED** [review]
Moved to `scheduling`

**Pull request merged**